File: multifiledownloader/src/main/java/com/github/why168/multifiledownloader/ICallback.java
package com.github.why168.multifiledownloader;

/**
 * @author Edwin.Wu
 * @version 2017/6/29 11:47
 * @since JDK1.8
 */
public interface ICallback {

    void onConnection();

    void onProgress();

    void onPaused();

    void onCompleted();

    void onError();

    void onRetry();
}

File: multifiledownloader/src/main/java/com/github/why168/multifiledownloader/ICallback.kt
package com.github.why168.multifiledownloader

/**
 * @author Edwin.Wu
 * @version 2017/6/29 11:47
 * @since JDK1.8
 */
interface ICallback {
    fun onConnection()
    fun onProgress()
    fun onPaused()
    fun onCompleted()
    fun onError()
    fun onRetry()
}